hypertension/dc-7101-algorithms/py-root/lib/utils.py

```diff
@@ -1,15 +1,146 @@
 from datetime import datetime
+from dateutil.relativedelta import relativedelta
+import operator
 
-# TODO: This might be overwriting whatever data object is passed in as bp_readings (I dont know b/c I dont know how Python works). If this is happening, change this function so that it doesnt overwrite the original data anymore.
-def add_proper_dates(bp_readings):
-    for bp_reading in bp_readings:
-        bp_reading.update({ "date_object": datetime.fromisoformat(bp_reading["date"]) })
+def calculate_correct_bp_reading(bp_readings, use_diastolic):
```

Comment:
This function could be broken into 2 or 3 (for clarity), and we don't have any tests for it, but I think that the surplus of tests for `sufficient_to_autopopulate` (where it is used) should cover our bases/was just trying to get a working PR up. However, I'm game for adding tests or splitting it into multiple functions if you think that would be more clear

Reply:
I can also add docstrings to help define the parameters and output, if you want us to do that. We could then use something like sphinx to generate docs, so we don't have to manually write them

Reply:
Tyler did this refactoring work.

(Resolved)